subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AvailableJavaHomes.java

```diff
@@ -190,113 +199,102 @@ public static File getBestJre() {
     }
 
     private static List<JvmInstallation> getJvms() {
-        if (jvms == null) {
-            NativeServices nativeServices = NativeServicesTestFixture.getInstance();
-            FileCanonicalizer fileCanonicalizer = nativeServices.get(FileCanonicalizer.class);
-            jvms = new ArrayList<>();
-            jvms.addAll(new DevInfrastructureJvmLocator(fileCanonicalizer).findJvms());
-            InstalledJvmLocator installedJvmLocator = new InstalledJvmLocator(OperatingSystem.current(), Jvm.current(), nativeServices.get(WindowsRegistry.class), nativeServices.get(SystemInfo.class), fileCanonicalizer);
-            jvms.addAll(new DevInfrastructureJvmLocator(fileCanonicalizer).findJvms());
-            jvms.addAll(installedJvmLocator.findJvms());
-            if (OperatingSystem.current().isLinux()) {
-                jvms.addAll(new BaseDirJvmLocator(fileCanonicalizer, new File("/opt")).findJvms());
-            }
-            if (!OperatingSystem.current().isWindows()) {
-                jvms.addAll(new SdkManJvmLocator(fileCanonicalizer).findJvms());
-            }
-            jvms.addAll(new HomeDirJvmLocator(fileCanonicalizer).findJvms());
-            jvms.addAll(new EnvVariableJvmLocator().findJvms());
-            // Order from most recent to least recent
-            Collections.sort(jvms, (o1, o2) -> o2.getVersion().compareTo(o1.getVersion()));
-        }
+        return jvms.get();
+    }
+
+    private static List<JvmInstallation> discoverLocalInstallations() {
```

Comment:
So the scope is really the detection.
Do you think it makes sense to also rework the used types to leverage the toolchain types?

Reply:
Oh yes. I think we have another 2 angles: the services and once those are aligned, we can align the different data structures representing installations (`Jvm`, `JavaInstallation`, `JavaInfo`, ...). Tracked on #14030 for 6.8